src/testkit/scala/tools/testkit/AssertUtil.scala

```diff
@@ -370,11 +370,15 @@ class NoTrace[A](body: => A) extends Runnable {
     val group = new ThreadGroup("notrace") {
       override def uncaughtException(t: Thread, e: Throwable): Unit = synchronized {
         uncaught += ((t, e))
+        ()
```

Comment:
I used `synchronized[buf.type] { buf += x }` somewhere, but I think it didn't work here. Needs a stable path?